### DISCUSSION TITLE
Thanks!

### DISCUSSION THREAD

**Discussion by 2talltech** [nickcoutsos/keymap-editor#160] [General]
Just finished a wireless Contra build with a Nice!Nano v2 and found your keymap editor. It works! I've only just picked up ZMK and the learning curve is steep. Your keymap editor is a relief! Thank you for your effort.

I hope someday the keys which have several keycodes - like the top row of numbers/symbols - will show on the editor, rather than just the numbers. I'm not sure how to explain it. I know that I get '_' when I press Shift and '-', for example, but that's not visible. Maybe I haven't found that option yet?

Amazing work!

**Comment by nickcoutsos** [comment]
Glad you like it!

It's not something you're missing, the editor only displays the un-shifted keycode (except for keycodes which already represent the shifted version like `AT` which is effectively sent as `LS(N2)`.

To display that s well I think I'd have to add metadata to reference the shifted version of the code but I'd also have to think about how that affects rendering of the key... and then there's also consideration for other locales...

**Comment by TommyChester** [comment]
I dont want to clog your discussions with more thank you's but I wanted to say how awesome this is and how much frickin time it has saved me. I am a guy who loves a custom board and does not have time to make, I had Skree at https://www.etsy.com/shop/TheBigSkree?ref=profile_header make my board for me, and he pointed me at your set up and it has been really easy to iterate over my set up. Thank you!

**Reply by nickcoutsos** [reply]
Thanks Tommy, glad to hear that!
